src/app/screens/Accounts/Detail/index.tsx

```diff
@@ -148,20 +148,23 @@ function AccountScreen() {
         throw new Error("No account available");
       }
 
-      // Validate the private key before saving
-      nostrPrivateKey && generatePublicKey(nostrPrivateKey);
-      nostrPrivateKey && nostrlib.hexToNip19(nostrPrivateKey, "nsec");
-
-      await msg.request("nostr/setPrivateKey", {
-        id: account.id,
-        privateKey: nostrPrivateKey,
-      });
-
       if (nostrPrivateKey) {
+        // Validate the private key before saving
+        generatePublicKey(nostrPrivateKey);
```

Comment:
Why do you need to generate the the public key here?

Reply:
We use the useEffect hook to update the public key and private key (in npub and nsec from the hex private key string), but here we have to show the errors as this is the try block, so we do it again here, just for validation.